lib/modem_attest_token/modem_attest_token.c

```diff
@@ -35,87 +35,28 @@
 #define UUID_HYPHEN_POS_3 (16 / 2)
 #define UUID_HYPHEN_POS_4 (20 / 2)
 
-#define ATTEST_TOKEN_RESP_LEN 200
-BUILD_ASSERT(
-	ATTEST_TOKEN_RESP_LEN <= CONFIG_AT_CMD_RESPONSE_MAX_LEN,
-	"Max AT cmd response length is too small for attestation token response");
-
-static int parse_resp(char *const resp, char **attest, char **cose)
-{
-	if (!resp || !attest || !cose) {
-		return -EINVAL;
-	}
-
-	char *cose_end = NULL;
-	char *attest_end = NULL;
-
-	*cose = NULL;
-
-	*attest = strchr(resp, '"');
-	if (!(*attest)) {
-		return -EBADMSG;
-	}
-	/* Move beyond first quotation mark */
-	*attest = *attest + 1;
-
-	attest_end = strchr(*attest, '.');
-	if (!(*attest_end)) {
-		return -EBADMSG;
-	}
-
-	/* Replace . with a NULL */
-	*attest_end = '\0';
-
-	*cose = attest_end + 1;
-	cose_end = strchr(*cose, '"');
-	if (!cose_end) {
-		return -EBADMSG;
-	}
-
-	/* Replace " with a NULL */
-	*cose_end = '\0';
-
-	return 0;
-}
-
 int modem_attest_token_get(struct nrf_attestation_token *const token)
 {
-	if (!token) {
-		return -EINVAL;
-	} else if ((token->attest && !token->attest_sz) ||
-		   (token->cose && !token->cose_sz)) {
-		return -EBADF;
-	}
 
 	int ret = 0;
-	enum at_cmd_state state;
-	char *cmd_resp = NULL;
-	char *attest = NULL;
-	char *cose = NULL;
+	char attest[64] = {0};
+	char cose[64] = {0};
```

Comment:
Anybody knows how large can these really be?

Reply:
Example seems quite small https://infocenter.nordicsemi.com/topic/ref_at_commands/REF/at_commands/security/attesttoken_set.html

I'm not sure if these are fixed in size, Hasnain probably knows. (I'll try to find his Github account)

Reply:
64 is too small... this was causing my bus fault.
here is an example attestation token:
```
2dn3hQFQUFA2QjI5T4aAxAYSLimWIwNQVT1XNJ-9R96MMoajfNDqSVCMpoIK91tLoQb5CuTh6FxQ.0oRDoQEmoQRBIfZYQI9Q4JPVaEm5a7zNGIqz9AKnvWQ54L5aFStg-2lQvEnVBV5RJkprOTxrV8ZoHwG0VscGnmoAVDChjHVay4JcH3A
```
original code used:
`#define ATTEST_TOKEN_RESP_LEN 200`
for the response buffer

Reply:
thanks, have increased both buffers to 128 bytes